tests/unit/amo/components/TestCollection.js

```diff
@@ -661,7 +659,7 @@ describe(__filename, () => {
 
     const wrapper = renderComponent({ store });
 
-    expect(wrapper.find('.Collection-edit-link')).toHaveLength(1);
```

Comment:
We do want to change this test, but because you've changed the meaning of the test, you also need to update the description. Right now the test is still called `renders an edit link when user has 'Collections:Edit' permission`, but it should be `hides the edit link when user has 'Collections:Edit' permission`.